samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -261,7 +285,15 @@ class OffsetManager(
         val sspToOffsets = checkpoint.getOffsets
         if(sspToOffsets != null) {
           sspToOffsets.asScala.foreach {
-            case (ssp, cp) => offsetManagerMetrics.checkpointedOffsets.get(ssp).set(cp)
+            case (ssp, cp) => {
+              offsetManagerMetrics.checkpointedOffsets.get(ssp).set(cp)
+
+              // Delete startpoints once a new checkpoint is committed.
+              startpoints.get(taskName) match {
+                case Some(_) => startpointManager.deleteStartpointForTask(ssp, taskName)
```

Comment:
Do we need to check here if the ssp was defined in a startpoint for a task before deleting it. Let's say there're 1000 ssps mapped to a task in the commit and  the startpoint is defined for only one SSP of the task. Wouldn't this  make unnecessary delete calls?